### PR TITLE
Add tip for GitHub Private Pages

### PR DESCRIPTION
**Pull request by alvgaona**
#### Description

The current documentation does not completely take into account private GitHub Pages within organizations. GitHub will create a random URL that needs a different treatment.

### CODE REVIEW

File: src/content/docs/en/guides/deploy/github.mdx [hunk 1]
```diff
@@ -42,6 +42,17 @@ Astro maintains the official `withastro/action` to deploy your project with very
...
 
+    :::tip[GitHub Organizations]
+    If you're not using custom domains for your [private pages](https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site),
+    GitHub will create a random URL for you. 
+
+    ```text
+    https://<random-string>.pages.github.io/
+    ```
+
+    Make sure you configure the `site` key with the given URL - no `base` needed.
```
Comment (dreyfus92) [inline src/content/docs/en/guides/deploy/github.mdx:53]:
```suggestion
    Make sure to set the `site` key to match the auto-generated URL. There's no need to specify a `base` key.
```

**Review by dreyfus92 — CHANGES_REQUESTED** [review]
Hey @alvgaona, thanks for tackling this 😁. I left a small suggestion.

File: src/content/docs/en/guides/deploy/github.mdx (hunk 1 above)
Comment (alvgaona) [reply src/content/docs/en/guides/deploy/github.mdx]: Awesome. Thank you!

File: src/content/docs/en/guides/deploy/github.mdx [hunk 2]
```diff
@@ -42,6 +42,17 @@ Astro maintains the official `withastro/action` to deploy your project with very
...
 
+    :::tip[GitHub Organizations]
+    If you're not using custom domains for your [private pages](https://docs.github.com/en/enterprise-cloud@latest/pages/getting-started-with-github-pages/changing-the-visibility-of-your-github-pages-site),
+    GitHub will create a random URL for you. 
+
+    ```text
+    https://<random-string>.pages.github.io/
+    ```
+
+    Make sure to set the `site` key to match the auto-generated URL. There's no need to specify a `base` key.
```
Comment (lilnasy) [inline src/content/docs/en/guides/deploy/github.mdx:53]: What happens when the `site` config is not set or set to an incorrect origin?

**Comment by sarah11918** [comment]
Hi @alvgaona ! This is just a quick note to let you know I'll be looking at this submission this week!

I know it might seem like a small addition, and maybe shouldn't need much review, but even though you've only added a "tip"... when you look at your submission in the surrounding context of the page, it looks like this:

notecautiontip

This... does not make for good docs.  :sweat_smile: 

What this tells me is that we need to rework our actual documentation with better guidance on using `base`... just adding on more notes, cautions and tips is not the solution here. This is why I haven't yet had the time to review your PR (it's not a "quick win" for me during a busy launch week) and clearly is an indication of much bigger docs work that needs to happen. 

But, it is on my plate to address this! Thank you for your patience!

**Comment by sarah11918** [comment]
OK! Thank you for your patience @alvgaona !  :raised_hands: 

There were just too many notes and cautions etc. so that told me that this content needed to be restructured. In fact, I made sections for using a custom domain vs not, since that often determines what you do or don't have to set.

I did of course include the information you provided, just in a different place/way. Thank you for contributing to this page!

**Comment by alvgaona** [comment]
> OK! Thank you for your patience @alvgaona ! 🙌
> 
> There were just too many notes and cautions etc. so that told me that this content needed to be restructured. In fact, I made sections for using a custom domain vs not, since that often determines what you do or don't have to set.
> 
> I did of course include the information you provided, just in a different place/way. Thank you for contributing to this page!

Thank you, @sarah11918! I

**Pull request merged**